syntax = "proto3";

package alerts;

service AlertService {
    rpc CountAllertOn(CountAlertRequest) returns (CountAlertResponse);
    rpc CountAllertOff(CountAlertRequest) returns (CountAlertResponse);
    rpc SendImageAlertOn(SendImageAlertRequest) returns (SendImageAlertResponse);
    rpc SendImageAlertOff(SendImageAlertRequest) returns (SendImageAlertResponse);
    rpc AccidentAlertOn(AccidentAlertRequest) returns (AccidentAlertResponse);
    rpc AccidentAlertOff(AccidentAlertRequest) returns (AccidentAlertResponse);
    rpc OddEventAlertOn(OddEventAlertRequest) returns (OddEventAlertResponse);
    rpc OddEventAlertOff(OddEventAlertRequest) returns (OddEventAlertResponse);
    rpc IsEmptyAlertOn(IsEmptyAlertRequest) returns (IsEmptyAlertResponse);
    rpc IsEmptyAlertOff(IsEmptyAlertRequest) returns (IsEmptyAlertResponse);
    rpc SetCountResult(SetCountResultRequest) returns (SetCountResultResponse);
    rpc SetAccidentResult(SetAccidentResultRequest) returns (SetAccidentResultResponse);
    rpc SetSendImageResult(SetSendImageResultRequest) returns (SetSendImageResultResponse);
    rpc SetOddEventResult(SetOddEventResultRequest) returns (SetOddEventResultResponse);
    rpc SetIsEmptyResult(SetIsEmptyResultRequest) returns (SetIsEmptyResultResponse);
    rpc GetCountResult(GetCountResultRequest) returns (GetCountResultResponse);
    rpc GetAccidentResult(GetAccidentResultRequest) returns (GetAccidentResultResponse);
    rpc GetSendImageResult(GetSendImageResultRequest) returns (GetSendImageResultResponse);
    rpc GetOddEventResult(GetOddEventResultRequest) returns (GetOddEventResultResponse);
    rpc GetIsEmptyResult(GetIsEmptyResultRequest) returns (GetIsEmptyResultResponse);
}

message CountAlertRequest {
    bool is_on = 1;
    int32 count = 2;
    int32 coordinate1_x = 3;
    int32 coordinate1_y = 4;
    int32 coordinate2_x = 5;
    int32 coordinate2_y = 6;
    bytes image = 7;
}

message CountAlertResponse {
    string message = 1;
}

message SendImageAlertRequest {
    bool is_on = 1;
}

message SendImageAlertResponse {
    string message = 1;
}

message AccidentAlertRequest {
    bool is_on = 1;
}

message AccidentAlertResponse {
    string message = 1;
}

message OddEventAlertRequest {
    bool is_on = 1;
}

message OddEventAlertResponse {
    string message = 1;
}

message IsEmptyAlertRequest {
    bool is_on = 1;
}

message IsEmptyAlertResponse {
    string message = 1;
}
message SetCountResultRequest{
    int32 count=1;
}
message SetAccidentResultRequest{
    bool accident=1;
}
message SetSendImageResultRequest{
    bytes image=1;
}
message SetOddEventResultRequest{
    bool odd_event=1;
}
message SetIsEmptyResultRequest{
    bool is_empty=1;
}
message SetCountResultResponse{
    string message=1;
}
message SetAccidentResultResponse{
    string message=1;
}
message SetSendImageResultResponse{
    string message=1;
}
message SetOddEventResultResponse{
    string message=1;
}
message SetIsEmptyResultResponse{
    string message=1;
}
message GetCountResultRequest{
    
}
message GetAccidentResultRequest{
}
message GetSendImageResultRequest{
}
message GetOddEventResultRequest{
}
message GetIsEmptyResultRequest{
}
message GetCountResultResponse{
    int32 count=1;
}
message GetAccidentResultResponse{
    bool accident=1;
}
message GetSendImageResultResponse{
    bytes image=1;
}
message GetOddEventResultResponse{
    bool odd_event=1;
}
message GetIsEmptyResultResponse{
    bool is_empty=1;
}
